frontend/src/components/parties/RelationshipProfile.js

```diff
@@ -102,11 +104,55 @@ export class RelationshipProfile extends Component {
 
   render() {
     const { id, typeCode } = this.props.match.params;
-    const mine = this.props.mines[id];
     const isPermittee = typeCode === "PMT";
-    const columnCount = 3 + (isPermittee ? 1 : 0);
-    // 24 is the total span from Ant Design
-    const width = 24 / columnCount;
+    const mine = this.props.mines[id];
+    const permitColumn = isPermittee
+      ? [
+          {
+            title: "Permit",
+            dataIndex: "permit",
+            render: (text) => <div title="Permit">{text}</div>,
+          },
+        ]
+      : [];
+    const columns = [
+      {
+        title: "Contact",
+        dataIndex: "contact",
+        render: (text, record) => (
+          <div title="Contact">
+            <Link to={router.PARTY_PROFILE.dynamicRoute(record.partyGuid)}>{text}</Link>
+          </div>
+        ),
+      },
+      {
+        title: "Role",
+        dataIndex: "role",
+        render: (text) => <div title="Role">{text}</div>,
+      },
+      ...permitColumn,
+      {
+        title: "Dates",
+        dataIndex: "dates",
+        render: (text, record) => (
+          <div title="Dates">
+            {record.startDate} - {record.endDate}
+          </div>
+        ),
+      },
+    ];
+
+    const transformRowData = (historyRelationships) =>
+      historyRelationships.map((relationship) => ({
+        key: relationship.mine_party_appt_guid,
+        contact: relationship.party.name,
+        partyGuid: relationship.party.party_guid,
+        role: this.state.partyRelationshipTitle,
+        permit: this.state.permitsMapping[relationship.related_guid],
+        endDate:
+          relationship.end_date === "9999-12-31" ? "Present" : formatDate(relationship.end_date),
```

Comment:
Same comment here about `null` values

Reply:
The end_date is `null` you mean? instead of `9999-12-31` ?

Reply:
That's right. James updated ETL to no longer set `mine_party_appt` `end_date`s to "9999-12-31". Mine manager `mine_party_appt`s `end_date`s are now set to `null` instead. We'll need to handle both for the time being

Reply:
updated!